libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxHybridSpeechPlayer.kt

```diff
@@ -0,0 +1,91 @@
+package com.mapbox.navigation.ui.voice.api
+
+import android.content.Context
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayer
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayerCallback
+import com.mapbox.navigation.ui.base.model.voice.Announcement
+import com.mapbox.navigation.ui.base.model.voice.SpeechState
+import java.util.Queue
+import java.util.concurrent.ConcurrentLinkedQueue
+
+/**
+ * Hybrid implementation of [SpeechPlayer] combining [MapboxOnboardSpeechPlayer] and
+ * [MapboxOffboardSpeechPlayer] speech players.
+ * @property context Context
+ * @property accessToken String
+ * @property language String
+ */
+class MapboxHybridSpeechPlayer(
+    private val context: Context,
+    private val accessToken: String,
+    private val language: String
+) : SpeechPlayer {
+
+    private val queue: Queue<SpeechState.Play> = ConcurrentLinkedQueue()
+    private val offboardSpeechPlayer: MapboxOffboardSpeechPlayer =
+        MapboxOffboardSpeechPlayer(context, accessToken, language)
+    private val onboardSpeechPlayer: MapboxOnboardSpeechPlayer =
+        MapboxOnboardSpeechPlayer(context, language)
+    private var clientCallback: SpeechPlayerCallback? = null
```

Comment:
`clientCallback` should be paired with the internal `play` request, we cannot store it as a single field, it's going to be overwritten by queued announcments.